src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,387 @@
+
+    // Target isotope
+    IsotopeView const& target_;
+
+    // Mott coefficients for the target element
+    WentzelElementData const& element_data_;
+
+    //! Calculates the form factor from the scattered polar angle
+    inline CELER_FUNCTION real_type calculate_form_factor(real_type cos_t) const;
+
+    //! Helper function for calculating the flat form factor
+    inline CELER_FUNCTION real_type flat_form_factor(real_type x) const;
+
+    //! Calculate the nuclear form momentum scale
+    inline CELER_FUNCTION real_type nuclear_form_momentum_scale() const;
+
+    //! Calculate the screening coefficient R^2 for electrons
+    CELER_CONSTEXPR_FUNCTION real_type screen_r_sq_elec() const;
+
+    //! Calculate the squared momentum transfer to the target
+    inline CELER_FUNCTION real_type mom_transfer_sq(real_type cos_t) const;
+
+    //! Momentum coefficient used in the flat nuclear form factor model
+    CELER_CONSTEXPR_FUNCTION real_type flat_coeff() const;
+
+    //! Momentum prefactor used in exponential and gaussian form factors
+    inline CELER_FUNCTION real_type nuclear_form_prefactor() const;
+
+    //! Samples the scattered polar angle based on the maximum scattering
+    //! angle and screening coefficient.
+    template<class Engine>
+    inline CELER_FUNCTION real_type sample_cos_t(real_type screen_coeff,
+                                                 real_type cos_t_max,
+                                                 Engine& rng) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with state and data from WentzelInteractor
+ */
+CELER_FUNCTION
+WentzelDistribution::WentzelDistribution(ParticleTrackView const& particle,
+                                         IsotopeView const& target,
+                                         WentzelElementData const& element_data,
+                                         real_type cutoff_energy,
+                                         WentzelRef const& data)
+    : data_(data)
+    , particle_(particle)
+    , cutoff_energy_(cutoff_energy)
+    , target_(target)
+    , element_data_(element_data)
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Samples the final direction of the interaction. This direction is in
+ * the frame where the incident particle's momentum is oriented along the
+ * z-axis, so it's final direction in the lab frame will need to be rotated.
+ */
+template<class Engine>
+CELER_FUNCTION Real3 WentzelDistribution::operator()(Engine& rng) const
+{
+    real_type screen_coeff = compute_screening_coefficient();
+    real_type cos_t_max_elec = compute_max_electron_cos_t();
+
+    // Randomly choose if scattered off of electrons instead
+    WentzelXsCalculator calc_electron_prob(
+        target_.atomic_number(), screen_coeff, cos_t_max_elec);
```

Comment:
Since these values don't depend on the rng, you can precalculate them in the constructor.